src/component/Board.jsx

```diff
@@ -1,46 +1,63 @@
-import { useState } from "react";
+import { useState, useEffect } from "react";
 import Card from "./Card";
 
-const cardValue = [];
-const cardLevel = 8;
-let card = [];
+function Board() {
+  const cardLevel = 20;
 
-// creating values of the card
-const createCardValue = (cardLevel) => {
-  for (let i = 1; i <= cardLevel; i++) {
-    cardValue.push(Math.ceil(i / 2));
-  }
-};
+  // initialinzing state
+  const [cardDeck, setCardDeck] = useState([]);
+  const [compareCardArr, setCompareCardArr] = useState({});
+  const [gameOver, setGameOver] = useState(false);
+  const [heading, setHeading] = useState(false);
+  const [pairCounter, setPairCounter] = useState(0);
 
-// shuffle the cardValue array
-function shuffle(array) {
-  array.sort(() => Math.random() - 0.5);
-}
+  useEffect(() => {
+    const cardValue = [];
+    let card = [];
 
-// creating card
-const createCards = (cardValue) => {
-  for (let i = 0; i < cardValue.length; i++) {
-    card.push({
-      flip: false,
-      value: cardValue[i],
-      id: i, // id updated
-    });
-  }
-};
+    // creating values of the card
+    const createCardValue = (cardLevel) => {
+      for (let i = 1; i <= cardLevel; i++) {
+        cardValue.push(Math.ceil(i / 2));
+      }
+    };
 
-const createGame = () => {
-  createCardValue(cardLevel);
-  shuffle(cardValue);
-  createCards(cardValue);
-};
+    // shuffle the cardValue array
+    function shuffle(array) {
+      array.sort(() => Math.random() - 0.5);
+    }
 
-createGame();
+    // creating card
+    const createCards = (cardValue) => {
+      for (let i = 0; i < cardValue.length; i++) {
+        card.push({
+          flip: false,
+          value: cardValue[i],
+          id: i,
+        });
+      }
+    };
```

Comment:
We can have this function outside of function component scope since it's not using any of the variable & it's a pure function

It will reduce initialisation of funtion

Reply:
implemented